src/main/cpp/alias/Alias.cpp

```diff
@@ -0,0 +1,77 @@
+/*
+ * Alias.cpp
+ *
+ *  Created on: Mar 11, 2017
+ *      Author: cedric
+ */
+
+#include "Alias.h"
+#include "AliasUtil.h"
+
+#include <exception>
+#include <assert.h>
+#include <math.h>
+
+namespace stride {
+namespace alias {
+
+Alias& Alias::CreateDistribution(std::vector<double> probabilities, util::Random& rng)
```

Comment:
Please change this to a regular `Alias` value that is constructed by this function and use a move constructor to return it to the caller.